### ISSUE TITLE
Package | Components architecture (VanillaJS/HTML)

### ISSUE THREAD

**Issue by jeffchew** [jeffchew/test-repo#46]
_Jeff-Chew created the following on May 23:_  

 ### Overview

This is the creation of the package architecture for the Components (VanillaJS/HTML flavor). This will include all IBM.com related components. 

### Additional Information
- Should look into the current `carbon-components` to see what architecture can be lifted from there

### Acceptance criteria
- [ ] Components should reference the `styles` package for CSS styling
- [ ] Storybook HTML as an output, as it will serve as the living documentation for this package
- [ ] Appropriate lint/unit testing functionality
- [ ] Pre-commit/pre-push hooks
- [ ] Publish scripts to generate modular output
- [ ] README on setup and usage
 

_Original issue: https://github.ibm.com/webstandards/digital-design/issues/860_

**Comment by jeffchew** [comment]
_On Jun 25, Jeff-Chew commented:_
@vladislav-saling @Kenny-Lam @james-dow I added in a story to look into Storybook for vanilla JS (https://github.ibm.com/webstandards/digital-design/issues/1120), which I think I'd prefer if it weren't for the Drupal team's ask for twig/json files. But we would also need a way to test the twig/json files themselves, which currently only see that support in pattern lab.

**Comment by jeffchew** [comment]
_On Jun 25, james-dow commented:_
@Jeff-Chew what type of testing are you expecting? Does it go beyond unit testing, or are you looking for interaction/behavior basic QA testing?

**Comment by jeffchew** [comment]
_On Jun 25, Jeff-Chew commented:_
@james-dow The latter. Testing meaning to make sure that the templates render properly. From what I can tell, storybook won't work with twig templates, but pure HTML as part of the stories. It only seems like pattern lab with the twig php engine would be able to render twig templates as part of the output, but I don't want to get tied down to using pattern lab because I honestly prefer Storybook. I'm torn because this is a must have request coming from the Drupal team.

cc: @Kenny-Lam @vladislav-saling @hahnrob @linda-carotenuto @Wonil-Suh1

**Comment by jeffchew** [comment]
_On Jun 25, james-dow commented:_
Correct me if I'm wrong, but we're talking about two separate issues, right?
  1. There is the QA testing process, and then...
  2. Drupal's oddly specific requirements.

In relation to Drupal, I haven't been a part of those discussions so I'm not totally looped into the requirements. But I am curious why Drupal wouldn't just use the given component/pattern's documentation page on the website? I'm a little confused why they would be special enough to warrant their own documentation site like Storybook.

Looping in @chsanche too.

**Comment by jeffchew** [comment]
_On Jun 26, Jeff-Chew commented:_
FYI, moving this out of the sprint for now. Want to have some conversations with Drupal team, as well as package approach internally before we move forward.
cc: @Kenny-Lam @james-dow @vladislav-saling @hahnrob

**Comment by jeffchew** [comment]
_On Jun 26, Jeff-Chew commented:_
I set up a meeting with our tech team and theirs in a couple of weeks. Hopefully we can hash this out. But the bottom line is that from an implementation standpoint, they don't need pattern lab.

What they asked for and need is a place to directly reference twig templates, where they would want us to provide those and they reference directly from our packages (and not from pattern lab, storybook, fractal, or any other automated templating tool).

Before the meeting with their team, I already added a topic in our morning huddle to talk more about this.
